### PR TITLE
Refactored tests, finished multi-Combinations, etc.

### LINKED ISSUE #14: Remove all instances of "exit"
Should keep naming consistent as `door` everywhere. This was my introduction initially, but there's no real reason for it, so I'm changing them to `door`.

### LINKED ISSUE #15: Make constants actually usable
Moving to only having a `constants` package and then having a class for each kind of constant within that. Turns out that one cannot declare `public static` constants within an inner class like I was trying to do before, i.e.

```java
public class Constants {
    public class Items {
        public static final Item BLUEBERRY = new Item(...)
    }
}
```

so now there is a package structure to do the same thing, e.g.

```
text-engine
...
|-> constants
    |-> Items.java
    |-> Effects.java
...
```

### LINKED ISSUE #16: Allow combinations between more than 2 items
Have the framework for this in place already.

Most of the work that still needs to be done is in `Item#combine(...)`.

### LINKED ISSUE #17: Add BaseTest class
Should have a simple setup of a `PlayerCharacter`, a few `Room`s connected by some locked and some unlocked `Door`s.

### LINKED ISSUE #19: Make constants more accessible, allow for easy copying
Add ` getCopyOf(Item item)` and `getCopiesOf(Item... items)` to `Items.java`, so that copies of constants can be easily fetched (should be used anywhere those constants are being added to a class to avoid having unintended side-effects on other `Item`s. Will be doing this with all of the constants classes as they grow.

### PR DESCRIPTION
**Pull request by ThoseGrapefruits**
A few different things in one PR. Sorry about the size, should have separated these out, but I was hitting a lot of stuff at the same time and it was easy to do a few refactors simultaneously.

<hr>

__Related Issues__
- Closes #14: Removed all instances of "exit". More naming consistency.
- Closes #15: Refactored constants into a package with top-level classes.
- Closes #16: Finished work for combining more than 2 items.
- Closes #17: Added BaseTest class with common testing environment goodness.
- Closes #19: Add `getCopyOf()` functions for `Item`s.

<hr>

I also encountered some bugs as I was adding tests, so I fixed them as I found them.

### CODE REVIEW

File: src/text_engine/boundaries/Door.java [hunk 1]
```diff
@@ -52,21 +52,22 @@ public Room getRoom2() {
      * @throws IllegalStateException door is locked
      * @throws IllegalArgumentException given room is not connected to this door
      */
-    public Room getOtherRoom(Room from) {
+    public Room getOtherRoom(Room from) throws IllegalArgumentException, IllegalStateException {
         Objects.requireNonNull(from);
```
Comment (dackJavies) [inline src/text_engine/boundaries/Door.java:56]: Pretty sure I was the one to drop these around, but for the sake of consistency, do we want to exclusively use @NotNull in the params for this?

**Comment by dackJavies** [comment]
nutte

File: src/text_engine/boundaries/Door.java (hunk 1 above)
Comment (ThoseGrapefruits) [reply src/text_engine/boundaries/Door.java:56]: No, `Objects.requireNonNull(from)` should always be used. `@NotNull` doesn't actually do any checks. It's a tip to compilers and IDEs that `null` shouldn't be passed in, and if it is, that they should get angry at the developer.

Comment (ThoseGrapefruits) [reply src/text_engine/boundaries/Door.java:56]: IntelliJ might actually be intelligent enough to check for `requireNotNull` checks and not need `@NotNull` decorators. I'll check.

**Pull request merged**